packages/block-editor/src/store/private-selectors.js

```diff
@@ -175,13 +176,29 @@ export function getOpenedBlockSettingsMenu( state ) {
 /**
  * Returns all style overrides, intended to be merged with global editor styles.
  *
+ * Overrides are sorted to match the order of the blocks they relate to. This
+ * is useful to maintain correct CSS cascade order.
+ *
  * @param {Object} state Global application state.
  *
  * @return {Map} A map of style IDs to style overrides.
  */
-export function getStyleOverrides( state ) {
-	return state.styleOverrides;
-}
+export const getStyleOverrides = createSelector(
+	( state ) => {
+		const clientIds = getClientIdsWithDescendants( state );
+		const clientIdMap = clientIds.reduce( ( acc, clientId, index ) => {
+			acc[ clientId ] = index;
+			return acc;
+		}, {} );
+
+		return [ ...state.styleOverrides ].sort( ( a, b ) => {
+			const aIndex = clientIdMap[ a[ 1 ].clientId ] ?? -1;
+			const bIndex = clientIdMap[ b[ 1 ].clientId ] ?? -1;
```

Comment:
What is a/b and what is at the other indices (other than 1)?

Reply:
Thanks for the nudge to make this clearer @ellatrix 👍 

I've pushed a commit that adds more meaningful names, along with an explanatory comment and fixed docblock comment. (https://github.com/WordPress/gutenberg/pull/61039/commits/8a2084066e5feff7053950e0e6946783235f5d1f)